### PR TITLE
First steps towards enabling TypeScript strict mode

### LINKED ISSUE #619: Enable TypeScript strict mode
## Description
[strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks) are currently not enabled, as they tend to increase boilerplate condition checks. It may lead to the code getting more verbose and being forced to check `null` and `undefined`, when we know for certain those values will not occur in certain scenarios.

However, considering that our contributors would be working on small improvements, it is better to have it enabled to avoid unforeseen bugs. The trade-off seems worth it.

### PR DESCRIPTION
**Pull request by seancolsen**
Works towards fixing #619 (but leaves much to be done later)

## Done

- [x] Fix lots of error that appear when strict mode is on.
- [x] Leave strict mode **off** for now. We'll turn it on in a later PR.
- [x] Use `undefined` instead of `null`.
- [x] Target the low-hanging-fruit, ignoring TS strict errors that require more thought and carful testing.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by seancolsen** [comment]
@pavish Here's a quick beginning to the process of making our typing more strict.

Making these changes across the codebase will be easy. We just need some clarity on on [how to deal with null vs undefined](https://github.com/centerofci/mathesar/discussions/825). The diff in 1776359f708ddf43613c1818847ecb1841e6698e will give you a sense of the kinds of changes we'll need to make. In lots of these scenarios, we can choose to resolve ambiguity towards `null` or towards `undefined`. I'd like to get some clarity on a general pattern to follow before continuing. Please weigh in within the discussion.

**Comment by pavish** [comment]
@seancolsen I've responded to the discussion. My preference would be to use `undefined` and only use `null` when serialization is required.

**Comment by seancolsen** [comment]
@pavish this is ready for review now.

I kept this same PR in-place but stripped out all the complex changes, leaving only the simple ones that we should be safe to merge now, with strict mode off.

I still have the rest of my changes locally and will build another PR with them later to continue working towards closing #619.

**Comment by pavish** [comment]
@seancolsen Could you please update this PR to resolve the merge conflicts? I'd like to merge this in before the other current open PRs at the time of this comment.

**Review by pavish — CHANGES_REQUESTED** [review]
@seancolsen The PR looks good for the major part.

There is one major bug that needs to be fixed: Add schema modal does not open. Once this bug is fixed, we can merge this PR.

File: mathesar_ui/src/component-library/file-upload/FileUpload.svelte [hunk 1]
```diff
@@ -86,8 +89,8 @@
   }
 </script>
 
-<div class="file-upload" class:inprogress={fileUploads?.length > 0}>
-  {#if fileUploads?.length > 0}
+<div class="file-upload" class:inprogress={fileUploads && fileUploads.length > 0}>
```
[the remark this answers is not in the document]
Comment (seancolsen) [reply mathesar_ui/src/component-library/file-upload/FileUpload.svelte]: Not sure what I was thinking there :thinking: 

Thanks for pointing this out. Fixed in da834c8f216863153309d8030518f8abfbb6032a

File: mathesar_ui/src/pages/schemas/Schemas.svelte [hunk 2]
```diff
@@ -80,9 +80,9 @@
   </div>
 </main>
 
-{#if isAddModalOpen}
+{#if isAddModalOpen && activeSchema}
```
[the remark this answers is not in the document]
Comment (seancolsen) [reply mathesar_ui/src/pages/schemas/Schemas.svelte]: My bad thanks for catching this. Fixed in fe380c318a176f7bcf2cb9cdb87b0e4b4cdf1c3c where I simplified `AddEditSchema` a bit.

File: mathesar_ui/src/pages/schemas/AddEditSchema.svelte [hunk 3]
```diff
@@ -14,11 +14,11 @@
 
   export let isOpen = false;
   export let isEditMode = false;
-  export let schema: SchemaEntry = null;
+  export let schema: SchemaEntry;
```
[the remark this answers is not in the document]
Comment (seancolsen) [reply mathesar_ui/src/pages/schemas/AddEditSchema.svelte]: Also fixed in fe380c318a176f7bcf2cb9cdb87b0e4b4cdf1c3c

**Comment by seancolsen** [comment]
@pavish ready for re-review

**Review by pavish — APPROVED** [review]

**Comment by pavish** [comment]
@seancolsen Looks good to me.

**Pull request merged**